integrations/adk-py/src/braintrust_adk/__init__.py

```diff
@@ -13,35 +17,321 @@ def setup_braintrust(
+        @wraps(wrapped)
+        async def wrapper(instance: Any, *args: Any, **kwargs: Any):
+            parent_context = args[0] if len(args) > 0 else kwargs.get("invocation_context")
+
+            async def _trace():
+                with start_span(
+                    name=f"agent_run [{instance.name}]",
+                    type=SpanTypeAttribute.TASK,
+                    metadata={"parent_context": _try_dict(parent_context), **_omit(kwargs, ["parent_context"])},
+                ) as agent_span:
+                    last_event = None
+                    async with aclosing(wrapped(instance, *args, **kwargs)) as agen:
+                        async for event in agen:
+                            if event.is_final_response():
+                                last_event = event
+                            yield event
+                    if last_event:
+                        agent_span.log(output=_try_dict(last_event))
+
+            async with aclosing(_trace()) as agen:
+                async for event in agen:
+                    yield event
+
+        return wrapper
+
+    Agent.run_async = trace_run_wrapper(Agent.run_async)
+    Agent._braintrust_patched = True
+    return Agent
+
+
+def wrap_flow(Flow: Any):
+    if _is_patched(Flow):
+        return Flow
+
+    def trace_flow(wrapped: Any):
+        @wraps(wrapped)
+        async def wrapper(instance: Any, *args: Any, **kwargs: Any):
+            invocation_context = args[0] if len(args) > 0 else kwargs.get("invocation_context")
+
+            async def _trace():
+                with start_span(
+                    name=f"call_llm",
+                    type=SpanTypeAttribute.TASK,
+                    metadata=_try_dict(
+                        {
+                            "invocation_context": invocation_context,
+                            **_omit(kwargs, ["invocation_context"]),
+                        }
+                    ),
+                ) as llm_span:
+                    last_event = None
+                    async with aclosing(wrapped(instance, *args, **kwargs)) as agen:
+                        async for event in agen:
+                            last_event = event
+                            yield event
+                    if last_event:
+                        llm_span.log(output=_try_dict(last_event))
+
+            async with aclosing(_trace()) as agen:
+                async for event in agen:
+                    yield event
+
+        return wrapper
+
+    Flow.run_async = trace_flow(Flow.run_async)
+
+    def trace_call_llm(wrapped: Any):
+        @wraps(wrapped)
+        async def wrapper(instance: Any, *args: Any, **kwargs: Any):
+            invocation_context = args[0] if len(args) > 0 else kwargs.get("invocation_context")
+            llm_request = args[1] if len(args) > 1 else kwargs.get("llm_request")
+            model_response_event = args[2] if len(args) > 2 else kwargs.get("model_response_event")
```

Comment:
do we need to look in both `args` and `kwargs` here and in `trace_run_wrapper`? below in `trace_run_sync_wrapper` and `trace_run_async_wrapper` we only look at `kwargs`. might be something about google adk that i am not familiar with